client/src/components/Navbar/index.tsx

```diff
@@ -83,13 +86,27 @@ function Navbar() {
     return () => window.removeEventListener('scroll', handleScroll);
   }, []);
 
+  useEffect(() => {
+    socket.on('sendNotification', (data) => {
+      setNotification((prev: any) => [...prev, data]);
+    });
```

Comment:
cleanup your effect,
check [using socket with react](https://socket.io/how-to/use-with-react-hooks)